src/main/java/uk/gov/justice/digital/delius/service/ReferralService.java

```diff
@@ -81,6 +82,22 @@ public ReferralStartResponse startNsiReferral(final String crn,
         })).build();
     }
 
+    @Transactional
+    public ReferralEndResponse endNsiReferral(final String crn, final String contextName, final ContextlessReferralEndRequest request) {
+
+        final var nsi = getExistingMatchingNsi(crn, contextName, request.getSentenceId(), request.getContractType(), request.getStartedAt())
+            .orElseThrow(() -> new BadRequestException(format("Cannot find NSI for CRN: %s Sentence: %d and ContractType %s", crn, request.getSentenceId(), request.getContractType())));
```

Comment:
I don't think getExistingMatchingNsi() is currently using the sentenceId. This should be included in the search critera

Reply:
will do

Reply:
It's done as part of the repo retrieval ... phew